packages/vm/src/ffi.rs

```diff
@@ -1,6 +1,36 @@
 use snafu::Snafu;
 use std::fmt::Debug;
 
+pub type FfiResult<T> = core::result::Result<FfiSuccess<T>, FfiError>;
+
+/// A return element and the gas cost of this FFI call
+pub type FfiSuccess<T> = (T, GasInfo);
```

Comment:
Interesting!
So, this was really unintuitive for me, because i have a set of expectations about the shape of `XyzResult` types, but i see why you chose to do this.
I'm thinking, maybe it would be better to call this `MeteredFfiResult`, or similar? then you can keep the old definition for the couple places where you wrote `Result<Something, FfiError>` to avoid triggering this type
Another idea: `FfiResult<Metered<T>>`?

Reply:
> I'm thinking, maybe it would be better to call this `MeteredFfiResult`, or similar? then you can keep the old definition for the couple places where you wrote `Result<Something, FfiError>` to avoid triggering this type
> Another idea: `FfiResult<Metered<T>>`?

The drawback is it allows you to use a `FfiResult` without taking care of gas. But every call into the backend must cause some gas consumption.

Reply:
hmm, I see...

Reply:
I guess it's ok then. I'd just add a line of documentation over the type alias definition to make this decision more visible in the code.

Reply:
I'm happy for alternative solution. But I want the gas information mandatory such that you can never forget them.

Reply:
I'll let you know if i have a good idea. I now understand the design considerations.